packages/cli/commands/project/migrateApp.js

```diff
@@ -0,0 +1,190 @@
+
+  let projectExists = false;
+  try {
+    await fetchProject(accountId, projectName);
+    projectExists = true;
+  } catch (e) {
+    projectExists = false;
+  }
+
+  if (projectExists) {
+    logger.error(
+      i18n(`${i18nKey}.errors.projectAlreadyExists`, {
+        projectName,
+      })
+    );
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  trackCommandUsage('migrate-app', {}, accountId);
+  if (appId && projectName && projectLocation) {
+    logger.log('');
+    uiLine();
+    logger.log(uiBetaTag(i18n(`${i18nKey}.warning.title`), false));
+    logger.log(i18n(`${i18nKey}.warning.projectConversion`));
+    logger.log(i18n(`${i18nKey}.warning.appConfig`));
+    logger.log('');
+    logger.log(i18n(`${i18nKey}.warning.buildAndDeploy`));
+    logger.log('');
+    logger.log(i18n(`${i18nKey}.warning.existingApps`));
+    logger.log('');
+    logger.log(i18n(`${i18nKey}.warning.copyApp`));
+    uiLine();
+
+    const { shouldCreateApp } = await promptUser({
+      name: 'shouldCreateApp',
+      type: 'confirm',
+      message: i18n(`${i18nKey}.createAppPrompt`),
+    });
+
+    if (!shouldCreateApp) {
+      process.exit(EXIT_CODES.SUCCESS);
+    }
+
+    try {
+      SpinniesManager.init();
+
+      SpinniesManager.add('migrateApp', {
+        text: i18n(`${i18nKey}.migrationStatus.inProgress`),
+      });
+
+      const { id } = await migratePublicApp(accountId, appId, projectName);
+      const { status, project } = await poll(getMigrationStatus, accountId, id);
+      if (status === 'SUCCESS') {
+        const absoluteDestPath = path.resolve(getCwd(), projectLocation);
+        const { env } = getAccountConfig(accountId);
+        const baseUrl = getHubSpotWebsiteOrigin(env);
+
+        const zippedProject = await downloadProject(accountId, projectName, 1);
+
+        await extractZipArchive(
+          zippedProject,
+          projectName,
+          path.resolve(absoluteDestPath),
+          { includesRootDir: true }
+        );
+
+        SpinniesManager.succeed('migrateApp', {
+          text: i18n(`${i18nKey}.migrationStatus.done`),
+          succeedColor: 'white',
+        });
+        logger.log('');
+        uiLine();
+        logger.success(i18n(`${i18nKey}.migrationStatus.success`));
+        logger.log('');
+        logger.log(
+          uiLink(
+            i18n(`${i18nKey}.projectDetailsLink`),
+            `${baseUrl}/developer-projects/${accountId}/project/${project.name}`
+          )
+        );
+        process.exit(EXIT_CODES.SUCCESS);
+      }
+    } catch (e) {
+      SpinniesManager.fail('migrateApp', {
+        text: i18n(`${i18nKey}.migrationStatus.failure`),
+        failColor: 'white',
+      });
+      logApiErrorInstance(e.error, new ApiErrorContext({ accountId }));
+      process.exit(EXIT_CODES.ERROR);
+    }
+  }
+};
+
+exports.builder = yargs => {
+  yargs.options({
+    name: {
+      describe: i18n(`${i18nKey}.options.name.describe`),
+      type: 'string',
+    },
+    location: {
```

Comment:
nit (And something to run by Mark), but I'm partial to the `dest` option that the CMS commands are already using. It's shorted and slightly more informative IMO

Reply:
@markhazlewood